Src/ILGPU.Algorithms/MatrixOperations/MaskedMatrixProcessor.cs

```diff
@@ -0,0 +1,114 @@
+//                                   ILGPU Algorithms
+//                           Copyright (c) 2023 ILGPU Project
+//                                    www.ilgpu.net
+//
+// File: MaskedMatrixProcessor.cs
+//
+// This file is part of ILGPU and is distributed under the University of Illinois Open
+// Source License. See LICENSE.txt for details.
+// ---------------------------------------------------------------------------------------
+
+using ILGPU.Runtime;
+using ILGPU.Util;
+using System;
+using System.Collections.Generic;
+
+namespace ILGPU.Algorithms.MatrixOperations
+{
+    /// <summary>
+    /// A processor for masked matrices to efficiently operate on multiple matrix
+    /// instances in parallel to maximize occupancy.
+    /// </summary>
+    public class MaskedMatrixProcessor<T, TPredicate, TStride, TProcessor>
+        : ConcurrentStreamProcessor
+        where T : unmanaged
+        where TStride : struct, IStride2D
+        where TPredicate : struct, InlineList.IPredicate<Index2D>
+        where TProcessor : struct, IMaskedSparseMatrixProcessor<T>
+    {
+        #region Instance
+
+        /// <summary>
+        /// The internal masked matrix multiplier which contains pre-compiled kernels.
+        /// </summary>
+        private readonly MaskedSparseMatrixMultiplier<T, TPredicate, TStride>
+            matrixMultiplier;
+
+        /// <summary>
+        /// Constructs a new masked processor.
+        /// </summary>
+        /// <param name="accelerator">The parent accelerator.</param>
+        /// <param name="maxNumConcurrentStreams">
+        /// The maximum number of concurrent streams to use (if any).
+        /// </param>
+        /// <param name="streamProvider">
+        /// A custom stream provider function to construct specialized streams.
+        /// </param>
+        public MaskedMatrixProcessor(
+            Accelerator accelerator,
+            int maxNumConcurrentStreams = 0,
+            Func<Accelerator, AcceleratorStream> streamProvider = null)
+            : base(accelerator, maxNumConcurrentStreams, streamProvider)
+        {
+            matrixMultiplier = accelerator.CreateSparseTransposedMatrixMultiplierMasked<
+                T,
+                TPredicate,
+                TStride,
+                TProcessor>();
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Returns the current predicate to use (if any).
+        /// </summary>
+        public TPredicate? Predicate { get; set; }
+
+        #region Methods
+
+        /// <summary>
+        /// Multiplies the given matrices using the currently assigned predicate.
+        /// </summary>
+        /// <param name="stream">The current accelerator stream to use.</param>
+        /// <param name="aView">The dense input matrix a of shape MxK.</param>
+        /// <param name="bView">The sparse matrix b of shape NxK (will transpose).</param>
+        /// <param name="outView">A dense output matrix of shape of aView.</param>
```

Comment:
I don't think the description of outView is correct. Doesn't outView contain aView * transpose(bView) which would have a shape of MxN.
I think it would be good to give a description of what this function is doing.
e.g. This function computes aView * bView_T  using the given accelerator stream.